### PR TITLE
PSL: Datamodel loader does not need a struct

### PR DESCRIPTION
**Pull request by pimeys**
We save no state in the loader, and hopefully will not be. Also it looks much better if we just call `module::function` instead of `Struct.function`.

Opportunistic refactoring, while reading the code.

### CODE REVIEW

File: psl/psl-core/src/validate.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
-mod datasource_loader;
+pub mod datasource_loader;
```
Comment (tomhoule) [inline psl/psl-core/src/validate.rs:1]: why not `pub(crate)`?

File: psl/psl-core/src/validate/datasource_loader.rs [hunk 2]
```diff
@@ -15,233 +15,226 @@ const SCHEMAS_KEY: &str = "schemas";
...
-        ast_schema: &ast::SchemaAst,
-        preview_features: BitFlags<PreviewFeature>,
-        diagnostics: &mut Diagnostics,
-        connectors: crate::ConnectorRegistry,
-    ) -> Vec<Datasource> {
-        let mut sources = Vec::new();
+/// Loads all datasources from the provided schema AST.
+/// - `ignore_datasource_urls`: datasource URLs are not parsed. They are replaced with dummy values.
+/// - `datasource_url_overrides`: datasource URLs are not parsed and overridden with the provided ones.
+pub fn load_datasources_from_ast(
```
Comment (tomhoule) [inline psl/psl-core/src/validate/datasource_loader.rs:21]: I think this should also be `pub(crate)` if possible

**Review by tomhoule — APPROVED** [review]
100% here for this. Let's do GeneratorLoader next. I just think we should work to make this machinery as private as possible so we have a clear public API in PSL. I think it's already private, let's check.

**Comment by tomhoule** [comment]
Yep it's not in the public API: https://prisma.github.io/prisma-engines/doc/psl/

**Comment by pimeys** [comment]
> 100% here for this. Let's do GeneratorLoader next. I just think we should work to make this machinery as private as possible so we have a clear public API in PSL. I think it's already private, let's check.

Did that too.

**Review by tomhoule — APPROVED** [review]
awesome

**Pull request merged**